cpp/src/strings/split/split.cu

```diff
@@ -497,6 +1017,76 @@ std::unique_ptr<experimental::table> rsplit( strings_column_view const& strings,
                      mr, stream);
 }
 
+contiguous_split_record_result contiguous_split_record(
+    strings_column_view const& strings,
+    string_scalar const& delimiter = string_scalar(""),
+    size_type maxsplit = -1,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+    cudaStream_t stream = 0) {
+  CUDF_EXPECTS(delimiter.is_valid(), "Parameter delimiter must be valid");
+
+  size_type max_tokens = std::numeric_limits<size_type>::max();
+  if (maxsplit > 0) {
+    max_tokens = maxsplit + 1;  // makes consistent with Pandas
+  }
+  auto has_validity = strings.parent().nullable();
+
+  auto d_strings_column_ptr =
+    column_device_view::create(strings.parent(), stream);
+  if (delimiter.size() == 0) {
+    return contiguous_split_record_fn(
+      strings,
+      whitespace_token_reader_fn<true>{
+        *d_strings_column_ptr, max_tokens, has_validity},
+      whitespace_token_copier_fn<true>{*d_strings_column_ptr, has_validity},
+      mr, stream);
+  }
+  else {
+    string_view d_delimiter(delimiter.data(), delimiter.size());
+    return contiguous_split_record_fn(
+      strings,
+      token_reader_fn<true>{
+        *d_strings_column_ptr, d_delimiter, max_tokens, has_validity},
+      token_copier_fn<true>{*d_strings_column_ptr, d_delimiter, has_validity},
+      mr, stream);
+  }
+}
+
+contiguous_split_record_result contiguous_rsplit_record(
+    strings_column_view const& strings,
+    string_scalar const& delimiter = string_scalar(""),
+    size_type maxsplit = -1,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+    cudaStream_t stream = 0) {
+  CUDF_EXPECTS(delimiter.is_valid(), "Parameter delimiter must be valid");
+
+  size_type max_tokens = std::numeric_limits<size_type>::max();
+  if (maxsplit > 0) {
+    max_tokens = maxsplit + 1;  // makes consistent with Pandas
+  }
+  auto has_validity = strings.parent().nullable();
+
+  auto d_strings_column_ptr =
+    column_device_view::create(strings.parent(),stream);
+  if (delimiter.size() == 0) {
+    return contiguous_split_record_fn(
+      strings,
+      whitespace_token_reader_fn<false>{
+        *d_strings_column_ptr, max_tokens, has_validity},
+      whitespace_token_copier_fn<false>{*d_strings_column_ptr, has_validity},
+      mr, stream);
+  }
+  else {
+    string_view d_delimiter(delimiter.data(), delimiter.size());
+    return contiguous_split_record_fn(
+      strings,
+      token_reader_fn<false>{
+        *d_strings_column_ptr, d_delimiter, max_tokens, has_validity},
+      token_copier_fn<false>{*d_strings_column_ptr, d_delimiter, has_validity},
+      mr, stream);
+  }
+}
+
```

Comment:
The only difference between `contiguous_split_record` and `contiguous_rsplit_record` is 4 lines where you are passing the template parameter for `whitespace_token_copier_fn<false/true>` and `token_reader_fn<false/true>`. Please refactor this code into a common function and then that curries the template parameter. 

Also, I am not sure if this is too much work, but I would recommend not using `bool` as the non-type template parameter. It is not obvious at all without looking at the definition of the template what is does (aka `bool forward`). My recommendation would be to use a scoped enumerator, something like:
```cpp
enum class Dir {
    FORWARD,
    BACKWARD
};

...
    token_reader_fn<Dir::FORWARD> ...
```